src/tests/ftest/control/dmg_storage_query.py

```diff
@@ -25,49 +25,48 @@ class DmgStorageQuery(ControlTestBase):
     :avocado: recursive
     """
 
-    def setUp(self):
-        """Set up for dmg storage query."""
-        super().setUp()
-        self.targets = self.server_managers[-1].get_config_value('targets')
-        md_on_ssd = False
+    def get_bdev_info(self):
+        """Get information about the server storage bdev configuration.
 
-        self.log_step('Determining server storage config')
+        Returns:
+            list: a list of dictionaries including information for each bdev device included in the
+                storage configuration
+        """
+        targets = self.server_managers[-1].get_config_value('targets')
+        md_on_ssd = False
         bdev_tiers = 0
-        self.bdev_list = []
+        bdev_info = []
         for engine in self.server_managers[-1].manager.job.yaml.engine_params:
             for index, tier in enumerate(engine.storage.storage_tiers):
                 if tier.storage_class.value == 'ram':
                     md_on_ssd = True
                 elif tier.storage_class.value == 'nvme':
                     bdev_tiers += 1
-                    for device in tier.bdev_list.value:
-                        self.bdev_list.append(
-                            {'bdev': device, 'roles': tier.roles.value, 'tier': index})
+                    for item, device in enumerate(tier.bdev_list.value):
+                        bdev_info.append(
+                            {'bdev': device,
+                             'roles': tier.roles.value,
+                             'tier': index,
+                             'tgt_ids': list(range(item, targets, len(tier.bdev_list.value)))})
```

Comment:
Question - How do you feel about this syntax?
```suggestion
                        bdev_info.append({
                            'bdev': device,
                            'roles': tier.roles.value,
                            'tier': index,
                            'tgt_ids': list(range(item, targets, len(tier.bdev_list.value)))})
```